pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -522,6 +544,46 @@ func (h *NetworkUtilsHandler) DisableTXOffloadChecksum(ifaceName string) error {
 	return nil
 }
 
+func (h *NetworkUtilsHandler) CreateRADaemon(socketPath string, advertisementIfaceName string, ipv6CIDR string, currentRetry int) error {
+	c, err := net.Dial("unix", socketPath)
+	if err != nil {
+		log.Log.V(4).Warningf("could not connect to the unix domain socket: %v", err)
+
+		if currentRetry > 0 {
+			time.Sleep(time.Second)
+			return h.CreateRADaemon(socketPath, advertisementIfaceName, ipv6CIDR, currentRetry-1)
+		}
+	}
+	defer c.Close()
+	fdConn := c.(*net.UnixConn)
+
+	log.Log.Info("managed to connect to unix domain socket; will retrieve the opened FD from it")
+
+	expectedFDNumber := 1
+	fds, err := fd.Get(fdConn, expectedFDNumber, []string{})
+	if err != nil || len(fds) == 0 {
+		return fmt.Errorf("error importing the NDP connection: %v", err)
+	}
+
+	openedFD := fds[expectedFDNumber-1]
+	defer func() {
+		_ = openedFD.Close()
+	}()
+
+	raDaemon, err := ndp.RouterAdvertisementDaemonFromFD(openedFD, advertisementIfaceName, ipv6CIDR)
+	if err != nil {
+		return fmt.Errorf("failed to re-create the RouterAdvertisement daemon on virt-launcher: %v", err)
+	}
+
+	go func() {
+		if err := raDaemon.Serve(); err != nil {
+			log.Log.Criticalf("could not listen via the Router Advertisement daemon to incoming requests: %v", err)
```

Comment:
I believe we should panic in case of a critical error.
Please use the same [pattern](https://github.com/kubevirt/kubevirt/blob/7c8a44cbfc5ec074a2a93f67f93fbce5063a002f/pkg/virt-launcher/virtwrap/network/common.go#L362) used for dhcp server.
```
log.Log.Errorf("could not listen via the Router Advertisement daemon to incoming requests: %v", err)
panic(err)
```